ebean-migration/src/main/java/io/ebean/migration/MigrationRunner.java

```diff
@@ -136,15 +136,17 @@ private void runMigrations(LocalMigrationResources resources, MigrationTable tab
     if (table.isEmpty()) {
       LocalMigrationResource initVersion = getInitVersion();
       if (initVersion != null) {
-        // run using a dbinit script
+        // invoke using a dbinit script
         logger.info("dbinit migration version:{}  local migrations:{}  checkState:{}", initVersion, localVersions.size(), checkStateMode);
-        checkMigrations = table.runInit(initVersion, localVersions);
+        checkMigrations.clear();
+        checkMigrations.addAll(table.runInit(initVersion, localVersions));
         return;
       }
     }
 
     logger.info("local migrations:{}  existing migrations:{}  checkState:{}", localVersions.size(), table.size(), checkStateMode);
-    checkMigrations = table.runAll(localVersions);
+    checkMigrations.clear();
```

Comment:
And similarly these 2 lines.  Given checkMigrations is protected etc ... no need for this change right?

Reply:
From the Separation Of Concerns aspect, I personally would deliver a copy of the arraylist. If someone (actually you) would change the result of table.runAll also the migrationRunner breaks the contract with the implementation part of someone else. Also the GC can cleanup Table.class immediately. In terms of the extendable approach, this should be reliable. 

Let's assume: table.runAll returns an empty list now, MigrationRunner would return an empty list. Table class is used internally only. If you modify Table.runAll for what ever reason and it will return null, also the code that implements MigrationRunner must be modified to avoid NPE or you have to take care also modifying MigrationRunner to deliver the same result as before. Hence, you have to modify two classes if you change the behaviour of Table.runAll. The code would compile but the result is an NPE later on.

From the clean code aspect, this is not the preferred way. So, I would recommend to return a copy of the list and initialise the arraylist in constructor. Also, if MigrationRunner should be extendable, somebody does not want to look into Table.class to know what exactly would be returned here.

But, I have to change it to:

`Optional.ofNullable(table.runInit(initVersion)).ifPresent(checkMigrations::addAll);` 

Otherwise it would break either but in MigrationRunner itself.

Since you are the owner: Please let me know if I should revert this part also or if you like me to change it to the Optional part above.

Reply:
Just revert that bit thanks.